app/assets/javascripts/samples/sample_datatable.js.erb

```diff
@@ -1193,34 +1193,56 @@ function changeToEditMode() {
 
       cancelEditMode();
 
-      var self = $(this);
-      var li = self.closest('li');
-      var text = li.find('.text');
-      if ($(text).find('.modal-tooltiptext').length > 0) {
-        text = $(text).find('.modal-tooltiptext');
+      var li = $(this).closest('li');
+      var url = li.attr('data-edit-url') + "/electronic_sign_edit";
+      ajaxCallEvent();
+
+      function ajaxCallEvent(){
+        $.ajax({
+          url: url,
+          success: function() {
+            var text = li.find('.text');
+            if ($(text).find('.modal-tooltiptext').length > 0) {
+              text = $(text).find('.modal-tooltiptext');
+            }
+            var textEdit = li.find('.text-edit');
```

Comment:
pro trick :smile:  In functions in JS you declare your variables at the top of the function you can do like this 
`var text, textEdit, controls, controlsEdit,.....;` 
and then assign some values to it. Read this: https://www.w3schools.com/js/js_hoisting.asp